### PR TITLE
Update autogenerated headers

### PR DESCRIPTION
**Pull request by maleadt**
This PR updates our wrapper-generator scripts to use the latest Clang.jl, which has several advantages: quite a lot of development has been done upstream (requiring no patches anymore to get through the CUDA headers), the generated code is cleaner (only a single source file), with more options for customization (e.g. using strongly-typed opaque pointers, which has already exposed several bugs in our wrappers).

In addition, I'm using the above to update our wrappers to CUDA 11.8.

### CODE REVIEW

**Comment by maleadt** [comment]
The one thing I'd still like to add here, is automated support for wrapping the functions that are currently gated behind an `ifdef __cplusplus` check because they use `half`.

File: res/wrap/wrap.jl [hunk 1]
```diff
@@ -1,511 +1,219 @@
...
+    end
 
-    (name == "all" || name == "nvml") && process("nvml", "$cuda/nvml.h"; include_dirs=[cuda])
+    if name == "all" || name == "nvml"
+        wrap("nvml", ["$cuda/nvml.h"]; include_dirs=[cuda])
+    end
 
-    (name == "all" || name == "cupti") && process("cupti", "$cupti/cupti.h", "$cupti/cupti_profiler_target.h";
+    if name == "all" || name == "cupti"
+        # NOTE: libclang (the C API) doesn't support/expose the __packed__/aligned attributes,
```
Comment (Gnimuc) [inline res/wrap/wrap.jl]: The latest generator supports packed/aligned structs/unions by overloading field accessing methods.

Comment (maleadt) [reply res/wrap/wrap.jl]: Great!

**Comment by maleadt** [comment]
Did it manually for now. Let's hope NVIDIA switches to __fp16 and __bf16 (I filed an issue for that).

**Pull request merged**